src/transform.c

```diff
@@ -822,12 +822,17 @@ int transform_load(struct augeas *aug, struct tree *xfm) {
         // FIXME: Record an error and return 0
         return -1;
     }
+
     r = filter_generate(xfm, aug->root, &nmatches, &matches);
     if (r == -1)
         return -1;
     for (int i=0; i < nmatches; i++) {
         const char *filename = matches[i] + strlen(aug->root) - 1;
         struct tree *finfo = file_info(aug, filename);
+
+        if (file != NULL && STRNEQ(filename, file) != 0)
```

Comment:
The condition should really just be `(file != NULL && STRNEQ(filename, file)`, i.e. without the `!= 0` - the `STRXXX` macros from `internal.h` really return boolean values (which, thanks to C, is pretty impossible to notice)